File: python3/2379.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        s = 0
        e = 0
        w = 0
        n = len(blocks)
        while e < k:
            if blocks[e] == "W":
                w += 1
            e += 1
        minw = w
        while e < n:
            if blocks[e] == "W":
                w += 1
            if blocks[s] == "W":
                w -= 1
            if w < minw:
                minw = w
            e += 1
            s += 1
        return minw

File: java/2379.java
class Solution {
    public int minimumRecolors(String b, int k) {
        char blocks[] = b.toCharArray();
        int s = 0, e = 0, w = 0, n = blocks.length;
        while (e < k)
            if (blocks[e++] == 'W')
                w++;
        int minw = w;
        while (e < n) {
            if (blocks[e++] == 'W')
                w++;
            if (blocks[s++] == 'W')
                w--;
            if (w < minw)
                minw = w;
        }
        return minw;
    }
}
